### ISSUE TITLE
ExtOperatorUpdate::update() is parsing query for each document

### ISSUE THREAD

**Issue by apkar** [FoundationDB/fdb-document-layer#156]
This function gets executed part plan execution phase, but it tries to parse and figure out the query here in this function. This function gets executed for each and every document we scan, most of the work in this function could be pushed to the planning phase.

It's possible I am being naive and expecting this to be simpler than it is, but it is worth spending a bit of time on it later.

**Comment by dongxinEric** [comment]
I agree. The query parsing and validating part can be lifted into the constructor of `ExtOperatorUpdate` and `ExtReplaceUpdate`.

**Comment by senthil-db-expert** [comment]
@apkar Could you please assign this to me, we started working on it.
